vscode-dotnet-runtime-library/src/Acquisition/IDistroDotnetSDKProvider.ts

```diff
@@ -0,0 +1,173 @@
+import { CommandExecutor } from '../Utils/CommandExecutor';
+/* tslint:disable:no-any */
+
+/**
+ * This interface describes the functionality needed to manage the .NET SDK on a specific distro and version of Linux.
+ *
+ * @remarks We accept community contributions of this interface for each distro-version pair.
+ * All calls which require sudo must leverage the vscode/sudo library. We will not accept contributions that use other methods to gain admin privellege.
+ * Please see DotnetDistroVersion as well to add your version.
+ */
+export abstract class IDistroDotnetSDKProvider {
+
+    protected commandRunner : ICommandExecutor;
+    protected distroVersion : DistroVersionPair;
+    protected distroJson : any | null = null;
+
+    protected preinstallCommandKey = 'preInstallCommands';
+    protected installCommandKey = 'installCommand';
+    protected uninstallCommandKey = 'uninstallCommand';
+    protected updateCommandKey = 'updateCommand';
+    protected packageLookupCommandKey = 'packageLookupCommand';
+    protected currentInstallPathCommandKey = 'currentInstallPathCommand';
+    protected isInstalledCommandKey = 'isInstalledCommand';
+    protected expectedMicrosoftFeedInstallDirKey = 'expectedMicrosoftFeedInstallDirectory';
+    protected expectedDistroFeedInstallDirKey = 'expectedDistroFeedInstallDirectory';
+    protected installedSDKVersionsCommandKey = 'installedSDKVersionsCommand';
+    protected installedRuntimeVersionsCommandKey = 'installedRuntimeVersionsCommand';
+    protected currentInstallVersionCommandKey = 'currentInstallationVersionCommand';
+
+    protected distroVersionsKey = 'versions';
+    protected versionKey = 'version';
+    protected dotnetPackagesKey = 'dotnet';
+    protected sdkKey = 'sdk';
+    protected runtimeKey = 'runtime';
+    protected aspNetKey = 'aspnetcore';
+
+    constructor(distroVersion : DistroVersionPair, executor : ICommandExecutor | null = null)
+    {
+        this.commandRunner = executor ?? new CommandExecutor();
+        this.distroVersion = distroVersion;
+        // Hard-code to the upper path (lib/dist/acquisition) from __dirname to the lib folder, as webpack-copy doesn't seem to copy the distro-support.json
+        const distroDataFile = path.join(path.dirname(path.dirname(__dirname)), 'distro-data', 'distro-support.json');
+        this.distroJson = JSON.parse(fs.readFileSync(distroDataFile, 'utf8'));
+
+        if(!distroVersion || !this.distroJson || !((this.distroJson as any)[this.distroVersion.distro]))
+        {
+            const error = new DotnetAcquisitionDistroUnknownError(new Error('We are unable to detect the distro or version of your machine'));
+            throw error;
+        }
+    }
+
+    /**
+     * Run the needed command(s) to install the .NET SDK on the machine 'globally.'
+     * Return '0' on success.
+     * @param installContext
+     */
+    public abstract installDotnet(fullySpecifiedVersion : string): Promise<string>;
```

Comment:
Are all these methods actually used by other classes? The answer might be yes, just feels like a lot. If not, we should keep as many of them private/protected as possible.

Reply:
Yea, mostly by the LinuxGlobalInstaller and LinuxVersionResolver. Pretty much all of them are used by that! If typescript had the pattern of friends I'd consider using that